Geometry_Engine/Query/IsContaining.cs

```diff
@@ -709,6 +709,158 @@ public static bool IsContaining(this Cuboid cuboid, IGeometry geometry, bool acc
+        [Input("tolerance", "Distance tolerance to use to determine intersections.")]
+        [Output("isContaining", "True if the point is contained within the bounds of the panels, false if it is not for each point provided.")]
+        private static bool IsContaining(this List<Polyline> closedVolume, List<Plane> planes, BoundingBox boundingBox, Point point, bool acceptOnEdges = false, double tolerance = BH.oM.Geometry.Tolerance.Distance)
+        {
+            //Return if point is null even without checking boundingBox.IsContaining(point)
+            if (point == null)
+            {
+                BH.Engine.Base.Compute.RecordError("Cannot query if a collection of panels contains a point if the point is null.");
+                return false;
+            }
+
+            if (planes.Any(x => x == null))
+            {
+                BH.Engine.Base.Compute.RecordError("Cannot query if a provided plane is null.");
+                return false;
+            }
+
+            if (!BH.Engine.Geometry.Query.IsContaining(boundingBox, point, true, tolerance))
+                return false;
+
+            Vector rVec;
+            bool isInPlane = false;
+            do
+            {
+                int seed = 1;
+                 rVec = BH.Engine.Geometry.Create.RandomVector(seed); 
+                
+                isInPlane = false;
+                foreach (Plane p in planes)
+                    isInPlane = isInPlane || rVec.IsInPlane(p, tolerance);
+
+                if (isInPlane)
+                {
+                    seed += 1;
+                }
+            }
+            while (isInPlane);
+
+            //Use a length longer than the longest side in the bounding box. Change to infinite line?
+            Line line = new Line() { Start = point, End = point.Translate(rVec * (((boundingBox.Max - boundingBox.Min).Length()) * 10)) };
+
+            //Check intersections
+            List<Point> intersectPoints = new List<Point>();
+
+            for (int x = 0; x < planes.Count; x++)
+            {
+                Point intPt = PlaneIntersection(line, planes[x], true, tolerance);
+                if (intPt == null)
+                    continue;
+                Polyline pLine = closedVolume[x];
+
+                if  (BH.Engine.Geometry.Query.IsContaining(boundingBox, intPt, true, tolerance) && BH.Engine.Geometry.Query.IsContaining(pLine, new List<Point> { intPt }, false, tolerance))
+                    intersectPoints.Add(intPt);
+            }
+
+            bool isContained = intersectPoints.CullDuplicates().Count % 2 != 0;
+
+            if (!isContained && acceptOnEdges)
+            {
+                //Check the edges in case the point is on the edge of the BE
+                foreach (Polyline p in closedVolume)
+                {
+                    if (point.IsOnCurve(p, tolerance))
+                        isContained = true;
```

Comment:
You missed exit from the loop. You can make the code even shorter btw:
```C#
if (!isContained && acceptOnEdges)
    isContained  = closedVolume.Any(p => point.IsOnCurve(p, tolerance));

```

```suggestion
                    if (point.IsOnCurve(p, tolerance))
                    {
                        isContained = true;
                        break;
                    }
```

Reply:
Updated.